src/JuliennedArrays.jl

```diff
@@ -142,26 +146,44 @@ julia> input = reshape(1:8, 2, 2, 2)
  6  8
 
 julia> s = Slices(input, 1, 3)
-2-element Slices{SubArray{Int64,2,Base.ReshapedArray{Int64,3,UnitRange{Int64},Tuple{}},Tuple{Base.OneTo{Int64},Int64,Base.OneTo{Int64}},false},1,Base.ReshapedArray{Int64,3,UnitRange{Int64},Tuple{}},Tuple{True,False,True}}:
+2-element Slices{SubArray{$Int, 2}, 1}:
  [1 5; 2 6]
  [3 7; 4 8]
 
 julia> map(sum, s)
-2-element Array{Int64,1}:
+2-element Vector{$Int}:
  14
  22
 ```
 """
-Slices(
-    whole::AbstractArray{Item,NumberOfDimensions},
-    alongs::Int...,
-) where {Item,NumberOfDimensions} =
-    Slices(whole, in_unrolled(as_vals(alongs...), ntuple(Val, NumberOfDimensions)...)...)
-export Slices
+function Slices(whole::AbstractArray{T,N}, alongs::Int...) where {T,N}
+    any(x->x>N, alongs) && throw(ArgumentError("All alongs values $(alongs) should be less than $(N)"))
+    Slices(whole, in_unrolled(as_vals(alongs...), ntuple(Val, N)...)...)
+end
+
+function Base.showarg(io::IO, ::Slices{T,N}, toplevel) where {T,N}
+    print(io, "Slices{", basetype(T), "{", eltype(T), ", ", ndims(T), "}, ", N, "}")
+end
 
+# This is expected to be added to Julia (maybe under a different name)
+# Follow https://github.com/JuliaLang/julia/issues/35543 for progress
+basetype(T::Type) = Base.typename(T).wrapper
+basetype(T) = basetype(typeof(T))
+
+###
+# Align
+###
 struct Align{Item,Dimensions,Sliced,Alongs} <: AbstractArray{Item,Dimensions}
     slices::Sliced
     alongs::Alongs
+    function Align{T,N,S,A}(slices::S, alongs::A) where {T,N,S,A}
+        sz = size(first(slices))
+        all(x->sz==size(x), slices) || throw(ArgumentError("All sizes of slices should be the same."))
```

Comment:
This seems like it could have a significant performance cost? Maybe annotate with `@boundscheck`?

Reply:
It turns out that the main overhead here is `map(size, slices)`:

```julia
julia> Xs = Slices(rand(2, 3, 4, 5), 1);

julia> @btime map(size, $Xs);
  77.824 ns (1 allocation: 576 bytes)
```

I've commented this check so that no significant runtime overhead is introduced. But I do believe we need this check to prevent undefined behaviors, so I mark it as "TODO".